velox/functions/prestosql/window/LeadLag.cpp

```diff
@@ -111,6 +107,10 @@ class LeadLagFunction : public exec::WindowFunction {
             constantOffset->as<ConstantVector<int64_t>>()->valueAt(0);
         VELOX_USER_CHECK_GE(
             constantOffset_.value(), 0, "Offset must be at least 0");
+        // If the offset is 0 then it means always return the current row.
```

Comment:
If offset is 0 we should have a fast path that returns the same value as the current row by setting rowNumbers array to the same values as the row numbers. There isn't really a need to change the logic this way.